src/app/src/main/kotlin/org/imperial/mrc/hint/security/SecurityConfig.kt

```diff
@@ -65,17 +65,22 @@ class Session(private val webContext: WebContext, private val pac4jConfig: Confi
     fun getVersionId(): String
     {
         //Generate a new id if none exists
-        return (pac4jConfig.sessionStore.get(webContext, VERSION_ID) ?: generateNewVersionId()) as String
+        return (pac4jConfig.sessionStore.get(webContext, VERSION_ID) ?: setNewVersionId()) as String
```

Comment:
Not sure why codecev is complaining about this line, as I'm explicitly testing both cases here...

Reply:
I'm not super clear on what this line of code does. It looks like it runs a conditional on getting the versionId for the session but then doesn't return anything if there is a versionId for the session? I'm not familiar with kotlin syntax but should it not return the session versionId between `?` and `:`?

Reply:
This is the Elvis operator - it evaluates the first clause, and returns that if it's not null. If it is null, it evaluates and returns the second clause. Useful for situations like these, where you want to fetch some data which may or may not exist, and do something else if it doesn't. 
https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator
I think you're expecting it to behave like the tertiary operator in js, which it does look a bit like.